refactor(spacing): generate spacing atomic classes programatically

dancormier opened the pull request:
This PR expands on https://github.com/StackExchange/Stacks/pull/1652 and closes https://github.com/StackExchange/Stacks/issues/1653

---

Spacing class generation is prone to bugs and accidental omissions. This PR make generation of spacing (and incidentally, positioning) atomic classes and documentation more programmatic.

dancormier committed:
refactor(spacing): generate spacing atomic classes programatically

This fixes the bug referenced in https://github.com/StackExchange/Stacks/pull/1652

dancormier committed:
docs(spacing): separate position and spacing; refactor

dancormier committed:
Positioning cleanup

dancormier committed:
update snapshot

dancormier committed:
Relocate spacing-mixins.less

giamir commented on lib/atomic/__snapshots__/spacing.less.test.ts.snap:
@dancormier just glancing at this PR. What would you think about adding the less snapshot test directly in develop to start with to then have here a better diff of what have actually changed with your refactor?

dancormier replied:
Done in this PR https://github.com/StackExchange/Stacks/pull/1657

giamir commented:
This refactor makes a lot of sense to me. I just glanced at the PR and left a comment. I will do a full review once you mark the PR as ready for review. Thanks @dancormier

dancormier committed:
Merge branch 'develop' into dcormier/spacing-refactor

dancormier committed:
Add less mixin definitions; rename mixins

dancormier committed:
Improve position docs

giamir commented on lib/atomic/__snapshots__/spacing.less.test.ts.snap:
@dancormier Not perfect but at least we have a diff of what changed now. Let me know when this PR is ready for a final review.

dancormier replied:
It's ready for your review now but no rush on this one

giamir commented on docs/_data/spacing.json:
```diff
@@ -1,446 +1,109 @@
+                "responsive": true
+            },
+            {
+                "term": "mxn",
```
Looks like `myn` is missing from this table. It was an issue also before these changes but we should correc it as part of this PR.
<img width="815" alt="Screenshot 2024-03-01 at 14 46 19" src="https://github.com/StackExchange/Stacks/assets/6231616/351a8261-c2f4-4530-b110-44d66e434c31">

giamir committed:
refactor(spacing): simplify mixin and remove a couple of extra classes created by mistake

giamir reviewed (approved):
I took the liberty to push [this commit](https://github.com/StackExchange/Stacks/pull/1654/commits/01b59cfadc924b3c2f3b2d5872cbaba7e7b63b86) and made a small observation about the docs missing `myn`.

Thanks for doing this refactor. I am good to merge when you are. 🙏

dancormier committed:
docs(spacing): add myn margin class abbr table

Addresses https://github.com/StackExchange/Stacks/pull/1654#discussion_r1509040097
